src/igvfd/types/file_set.py

```diff
@@ -965,6 +966,31 @@ def workflows(self, request, files=None):
                         analysis_set_workflows_set.add(workflow)
         return sorted(list(analysis_set_workflows_set))
 
+    @calculated_property(
+        condition='input_file_sets',
+        schema={
+            'title': 'Targeted Genes',
+            'description': 'A list of genes targeted by the input measurement sets assays.',
+            'type': 'array',
+            'notSubmittable': True,
+            'uniqueItem': True,
+            'minItems': 1,
+            'items': {
+                'title': 'Targeted Gene',
+                'type': 'string',
+                'linkTo': 'Gene'
+            }
+        }
+    )
+    def targeted_genes(self, request, input_file_sets=None):
+        analysis_set_targeted_genes = set()
+        for input_file_set in input_file_sets:
+            if input_file_set.startswith('/measurement-sets/'):
```

Comment:
are we only considering measurement sets here not including e.g. intermediate analysis sets for now?

Reply:
you're right it should also account for that.